include/deal.II/hp/fe_collection.h

```diff
@@ -739,11 +739,27 @@ namespace hp
      * Exception
      *
      * @ingroup Exceptions
+     *
+     * @deprecated Use DoFCellAccessor::ExcNoDominatedFiniteElementOnChildren
+     * instead.
      */
-    DeclExceptionMsg(
-      ExcNoDominatedFiniteElementAmongstChildren,
-      "No FiniteElement has been found in your FECollection that is "
-      "dominated by all children of a cell you are trying to coarsen!");
+    class DEAL_II_DEPRECATED ExcNoDominatedFiniteElementAmongstChildren
+      : public dealii::ExceptionBase
+    {
+    public:
+      ExcNoDominatedFiniteElementAmongstChildren()
+      {}
+      virtual ~ExcNoDominatedFiniteElementAmongstChildren() noexcept
+      {}
+      virtual void
+      print_info(std::ostream &out) const override
+      {
+        out << "    "
+            << "No FiniteElement has been found in your FECollection that is "
+               "dominated by all children of a cell you are trying to coarsen!"
+            << std::endl;
+      }
+    };
```

Comment:
I found that this exception should belong to the `DoFCellAccessor` class rather than to the `FECollection` itself. I could not think of a more elegant way to deprecate this exception. Should we delete it right away?

Reply:
Just nuke it. There is no need to deprecate exceptions unless you think that there is used code that catches this exception -- which I think is highly unlikely given that there is nothing user code could do if this exception is thrown.

Reply:
I can't think of any reason in which a user could have used that particular exception. I'll get rid of it.